<!doctype html>
<html class="no-js" lang="en" data-content_root="../../../">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="../../../genindex/" /><link rel="search" title="Search" href="../../../search/" />

    <link rel="shortcut icon" href="../../../_static/favicon.ico"/><!-- Generated with Sphinx 7.4.7 and Furo 2024.08.06 -->
        <title>Ancient DNA and LD - Arjun Biddanda</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=8f2a1f02" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/furo.css?v=354aac6f" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/tabs.css?v=4c969af8" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/furo-extensions.css?v=302659d7" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/override.css?v=e89897b8" />
    <link rel="stylesheet" type="text/css" href="https://use.typekit.net/ajj5weq.css" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --font-stack: acumin-pro, sans-serif;
  --color-background-primary: #f8f8f8;
  --color-background-secondary: #f8f8f8;
  --color-background-hover: #F5F3ED;
  --color-brand-primary: black;
  --color-brand-content: #A46F0D;
  --admonition-font-size: var(--font-size-normal);
  --admonition-title-font-size: var(--font-size-normal);
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>

<a class="skip-to-content muted-link" href="#furo-main-content">Skip to content</a>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../../../"><div class="brand">Arjun Biddanda</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
          <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><div class="sidebar-scroll"><a class="sidebar-brand" href="../../../">
  
  
  <span class="sidebar-brand-text">Arjun Biddanda</span>
  
</a><div class="sidebar-tree">
  <ul>
<li class="toctree-l1"><a class="reference internal" href="../../../research/">Research</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../cv/">Curriculum Vitae</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../publications/">Publications</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../software/">Software</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../">Blog</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../notes/">Notes &amp; Miscellanea</a></li>
</ul>

</div>
</div>
      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="view-this-page">
  <a class="muted-link" href="../../../_sources/blog/2021/2021-6-22-adna-ld.md.txt" title="View this page">
    <svg><use href="#svg-eye"></use></svg>
    <span class="visually-hidden">View this page</span>
  </a>
</div>
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
              <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main" id="furo-main-content">
          <section class="tex2jax_ignore mathjax_ignore" id="ancient-dna-and-ld">
<h1>Ancient DNA and LD<a class="headerlink" href="#ancient-dna-and-ld" title="Link to this heading">¶</a></h1>
<p>I wanted to take this time to highlight a couple of recent results from our paper looking at <a class="reference external" href="https://www.biorxiv.org/content/10.1101/2021.06.17.448867v1">time-stratified sampling and linkage disequilibrium in ancient DNA</a>. I should note that the post here assumes some familiarity with <a class="reference external" href="https://en.wikipedia.org/wiki/Coalescent_theory">coalescent theory</a> and I’ve tried my best to provide mostly intuitive reasoning behind some of our main findings.</p>
<section id="inspiration-and-starting-points">
<h2>Inspiration and Starting Points<a class="headerlink" href="#inspiration-and-starting-points" title="Link to this heading">¶</a></h2>
<p>Linkage disequilibrium is a fundamental signal that we observe in population genomic data, providing signal for genome-wide association study resolution, estimating the extent of admixture in humans, and the extent of natural selection. A major point of inspiration for this research was from <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1212677/">Ohta and Kimura</a> and <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/12399406/">Gil McVean’s</a> previous work on expectations of linkage disequilibrium (LD), or the covariation in alleles at different loci. The work from McVean in particular was able to relate the covariances in coalescent times at two loci to an approximation of LD:</p>
<div class="math-wrapper docutils container">
<div class="math notranslate nohighlight">
\[ \mathbb{E}[r^2] \approx \sigma^2_d = \frac{Cov(t^{(1)}_{ij},t^{(2)}_{ij}) - 2Cov(t^{(1)}_{ij},t^{(2)}_{ik}) + Cov(t^{(1)}_{ij},t^{(2)}_{kl})}{Cov(t_{ij}^{(1)},t_{kl}^{(2)})},\]</div>
</div>
<p>where <span class="math notranslate nohighlight">\(t^{(1)}_{ij}\)</span> is the coalescent time between haplotypes i and j at the first locus, and haplotypes i and j share the same allelic state. I’ve eliminated some of the subsequent math and nice derivation, but the key insight here is that we can draw the connection very generally between LD (an observable quantity in data) with properties of the genealogy at two loci (something about the process we don’t observe). A major benefit is that since the <em>genealogical properties are agnostic to the evolutionary model considered</em>, if we can compute the covariances in coalescent times under a model, then we can obtain approximations of LD. The scenarios explored in the current literature all dealt with samples from the same points in time, so we wanted to see if we could translate this to a case with samples from different points in time.</p>
<p>Another key model that is central to how researchers model haplotype data today is the haplotype-copying or <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/14704198/">Li and Stephens model</a>, which models a focal haplotype as a mosaic of other “template” haplotypes. The beauty of this approach is that you can approximate the full likelihood of the data (under mutation and recombination parameters), by looking at products of these conditional likelihoods, and it also has a natural interpretation of copying from a genealogical closest neighbor in the panel.</p>
<!-- <figure class="figure">
	<img src="/images/blog_images/adna_ld/listephens.png" width="700" height="350" alt="">
	<figcaption class="figcaption">Graphical depiction of haplotype-copying from a panel (based on a figure by <a href="https://doi.org/10.1093/bioinformatics/btx530">Zhu et al</a>).</figcaption>
</figure><br> -->
<a class="reference internal image-reference" href="../../../_images/listephens.png"><img alt="../../../_images/listephens.png" class="align-center" src="../../../_images/listephens.png" style="width: 700px; height: 350px;" />
</a>
<center>Graphical depiction of haplotype-copying from a panel <br/> (based on a figure by <a href="https://doi.org/10.1093/bioinformatics/btx530">Zhu et al</a>).</center>
<p>This is the standard model employed in imputation and phasing frameworks in statistical genetics to perform inference, so understanding its behavior in ancient samples is useful to set expectations for the models behavior at different time-depths (see <a class="reference external" href="https://www.nature.com/articles/ncomms6257">Gamba et al</a> for a nice empirical assessment of this problem though).</p>
<p>Therefore, the <strong>two key questions (at a high-level)</strong> we wanted to ask were:</p>
<ol class="arabic simple">
<li><p>Can we relate joint genealogical properties (e.g. covariance in coalescent times) in models with samples from different points in time  to observable LD statistics?</p></li>
<li><p>How does modeling an ancient focal sample with a modern panel behave under the haplotype copying model? Do we shift between copying from different template haplotypes more or less often?</p></li>
</ol>
<p>We provide an outline of our major results below.</p>
</section>
<section id="the-impact-of-time-separation-on-genealogical-history-at-two-loci">
<h2>The impact of time-separation on genealogical history at two loci<a class="headerlink" href="#the-impact-of-time-separation-on-genealogical-history-at-two-loci" title="Link to this heading">¶</a></h2>
<p>To keep this blogpost on the shorter side, we’ll start with the intuition under constant size and just two haplotype samples separated by <span class="math notranslate nohighlight">\(t_a\)</span> coalescent units. This consideration of a “two-locus, two-sample” system is the simplest type of system we could consider in models with recombination, and provides a nice set of analytical results as well. We found that time separation <em>increases</em> the expected branch length (at a single-locus) by a factor of <span class="math notranslate nohighlight">\(t_a\)</span> coalescent units, which is more branch length on which recombination events can occur*.</p>
<p>In Figure 1b of our paper we show that by increasing time-separation, for a constant recombination rate the probability of being in the “uncoupled” state at time <span class="math notranslate nohighlight">\(t_a\)</span> (where one lineage is ancestral to each of the two loci) is higher. The uncoupled state can only occur if a recombination event has occurred prior to <span class="math notranslate nohighlight">\(t_a\)</span> on the modern lineage, so this probability is a direct reflection of that.</p>
<!-- <figure class="figure">
	<img src="/images/blog_images/adna_ld/figure1b.png" width="300" height="300" alt="">
	<figcaption class="figcaption">The probability of lineages at two loci being "uncoupled" for a population-scaled recombination rate and sampling time.</figcaption>
</figure><br> -->
<a class="reference internal image-reference" href="../../../_images/figure1b.png"><img alt="../../../_images/figure1b.png" class="align-center" src="../../../_images/figure1b.png" style="width: 350px; height: 350px;" />
</a>
<center>The probability of lineages at two loci being<br/> "uncoupled" for a population-scaled <br/> recombination rate and sampling time.</center>
<p>This increase in the probability of lineages becoming uncoupled also affects joint patterns of variation such as the correlation in pairwise differences between two loci. Patterns of variation operate effectively under the same intuition - as time separation increases, the probability of recombination increases due to increased branch length, which then decreases the correlation in patterns of variation. <a class="reference external" href="https://www.biorxiv.org/content/10.1101/2021.06.17.448867v1">Figure 2 in our paper</a> verifies that we observe a decrease in the correlation in pairwise differences, and that the rate of decay as a function of recombination distance is also affected by <span class="math notranslate nohighlight">\(t_a\)</span> varying from <span class="math notranslate nohighlight">\(O(\rho^{-1})\)</span> with two modern samples to <span class="math notranslate nohighlight">\(O(\rho^{-2})\)</span> with very old samples, which we found to be an interesting mathematical result (see Figure 3 in our paper for more of an exploration of the correlation in pairwise differences between varying demographies.)</p>
<p>The other statistic that we focused our effort on is the normalized product of <span class="math notranslate nohighlight">\(D^{(0)}D^{(t)}\)</span>, where <span class="math notranslate nohighlight">\(D = p_{AB} - p_Ap_B\)</span> or the covariance in allelic states in a sample (the superscript refers to the timepoints). We derived how this product behaves in a model of constant population size:</p>
<div class="math-wrapper docutils container">
<div class="math notranslate nohighlight">
\[\sigma^2_t := \frac{(\rho + 2)(\rho + 10)}{(\rho^3 + 15\rho^2 + 48\rho + 48)e^{\frac{t_a(\rho+2)}{2}} - 4}\]</div>
</div>
<p>For confirmation, we set <span class="math notranslate nohighlight">\(t_a = 0\)</span> and we obtain the approximation for <span class="math notranslate nohighlight">\(\sigma^2_d\)</span> for two contemporary haplotypes (e.g. equation 12 in <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/12399406/">McVean 2002</a>). In simulations, we find that the product of LD is quite similar across different population scaled recombination rates (under the same population size) except for very large time-separations.</p>
<!-- <figure class="figure">
	<img src="/images/blog_images/adna_ld/figure5.png" width="300" height="300" alt="">
	<figcaption class="figcaption">The joint product of D across different time-stratified sampling levels (in generations).</figcaption>
</figure><br> -->
<a class="reference internal image-reference" href="../../../_images/figure5.png"><img alt="../../../_images/figure5.png" class="align-center" src="../../../_images/figure5.png" style="width: 350px; height: 350px;" />
</a>
<center>The joint product of D<br/> across different time-stratified <br/>sampling levels (in generations).</center>
</section>
<section id="copying-patterns-in-simulated-and-real-haplotype-data">
<h2>Copying Patterns in Simulated and Real Haplotype Data<a class="headerlink" href="#copying-patterns-in-simulated-and-real-haplotype-data" title="Link to this heading">¶</a></h2>
<p>When summarizing copying patterns, we had to choose a summary that was representative of the length of the copied blocks from the panel. This is important becuase it provides a genetic distance scale over which we can reliably expect to have accurate prediction of an unobserved variant due to linkage disequilibrium (e.g. genotype imputation). We focused most of our effort on the inferred “copying jump rate”, <span class="math notranslate nohighlight">\(\lambda\)</span>, in the haplotype copying model. We focused on this parameter because it is inversely related to the expected copying block length, <span class="math notranslate nohighlight">\(B\)</span> , where <span class="math notranslate nohighlight">\(\mathbb{E}[B] = 1/\hat{\lambda}\)</span> in Morgans.</p>
<p>We find that as the age of the focal sample increases, that in <em>most</em> cases the jump rate also increases, indicating that haplotype copying blocks generally get smaller for ancient samples (it’s useful to consider that <span class="math notranslate nohighlight">\(\lambda \sim 1000\)</span> means an average block-size of around 0.1 centimorgans, which is above the threshold for array capture products in ancient DNA).</p>
<p>One difference is in the case with extreme population growth, where we find that the copying jump rate actually decreases (making copied blocks longer) until the point of growth. We can think about how this might happen from intuition based on the extreme of a “star-like” genealogy (which extreme instantaneous growth should approximate). In a star-like genealogy, each mutation occurring will be unique to a sample making the number of shared mutations quite small and therefore the haplotype must be shared for longer in the copying model. For samples that are closer to the onset of growth there are lower opportunities for recombination to affect this ancient lineage since it has a shorter branch length, leading to the decrease of the copying jump rate as sample age increases.</p>
<!-- <figure class="figure">
	<img src="/images/blog_images/adna_ld/figure7.png" width="800" height="300" alt="">
	<figcaption class="figcaption">Results from the haplotype-copying model on actual SNP data and two simulated datasets with the same sample ages and demographies inferred from modern data.</figcaption>
</figure><br> -->
<a class="reference internal image-reference" href="../../../_images/figure7.png"><img alt="../../../_images/figure7.png" class="align-center" src="../../../_images/figure7.png" style="width: 800px; height: 300px;" />
</a>
<center>Results from the haplotype-copying model on actual SNP data and two simulated datasets with the same sample ages and demographies inferred from modern data.</center>
<p>We also sought to apply the copying model to real data from 334 ancient male X-chromosomes in publicly available ancient DNA data from Europe. We chose to use male X-chromosomes because they are haploid so we did not need to statistically phase them. However, we do clarify that this is a limitation of our approach and our interpretation is restricted to male-specific demography.</p>
<p>The most surprising finding in our analysis was that for samples within 500 generations that there is a substantial decrease in the haplotype copying jump rate in the real data that is not captured by simulating under previously inferred models of demographic history from modern samples (see above). We hypothesize in the paper that this could be due to an over-estimation of past population sizes due to fragmented populations and that it might warrant some further investigation. Indeed population structure can have the effect of creating excess heterozygosity (see Figure S6 in <a class="reference external" href="https://www.nature.com/articles/nature10231">Li &amp; Durbin</a> for an example of this). We shouldn’t be terribly surprised by this fact, after all ancient DNA has shown that historical population structure is quite substantial (and very interesting!), but this does provide a useful summary to look at potential discrepancies with inferences from modern genomes alone.</p>
</section>
<section id="outlook-future-work">
<h2>Outlook &amp; Future Work<a class="headerlink" href="#outlook-future-work" title="Link to this heading">¶</a></h2>
<p>There are a couple of interesting points that are natural next questions in this area that may (or may not) be useful:</p>
<ul class="simple">
<li><p>What modes of ancient population structure can lead to similar decreases in the copying jump rate? We didn’t explore this in the current paper as its asking a fundamentally different question on the consistency of specific models with data, but it would be nice to have an understanding of whether more realistic models of structure followed by growth could capture this effect.</p></li>
<li><p>Can the normalized product of <span class="math notranslate nohighlight">\(D^{(0)} D^{(t)}\)</span> across all pairs of time-points be used to estimate recent population demographic history? The data summary would be a <span class="math notranslate nohighlight">\(T\times T\times L\)</span> tensor of joint LD estimates across all timepoints at <span class="math notranslate nohighlight">\(L\)</span> different recombination bins, and one could employ a moment-based fitting procedure for the scaling factor of <span class="math notranslate nohighlight">\(N_e\)</span>. This might be more applicable in settings where there is no influence of population structure, but could be an interesting in-silico experiment.</p></li>
<li><p>In thinking about two-locus genealogies, we primarily wanted to see what we could do with an extension of the simple two-locus, two-sample model in a constant-sized population for two reasons: (1) the previous results are all analytical and (2) the system still provides insight on properties in data that we could observe. However, it’s not infeasible to think about how these two-locus systems might behave under <strong>both</strong> spatial and temporal sampling under an island or stepping-stone model. While analytical results might prove elusive and the model complexity increases exponentially with the number of demes, it may be possible that numerical results may still be obtainable. Also with recent insights into the applicaiton of multi-variate phase-type distributions in population genetics (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30822431/">Hobolth et al. 2019</a>)
, it could be possible to describe the full joint <strong>distribution</strong> of coalescent times rather than just joint <strong>moments</strong>.</p></li>
</ul>
</section>
<section id="personal-highlight">
<h2>Personal Highlight<a class="headerlink" href="#personal-highlight" title="Link to this heading">¶</a></h2>
<p>I’ve frequently enjoyed asking friends and colleagues what they’ve enjoyed most about a particular product of their work. Often what the authors find the most interesting or important doesn’t necessarily become the most important facet in the paper’s narrative. In this case, it was actually the material in Appendix 3 on computing the first coalescent time for an ancient sample into a panel of modern individuals. The time to first coalescence is a good measure of the branch length only subtending the ancient sample, which is useful for considering the expected number of private mutations and recombination events that can affect the ancient sample. This branch grows longer as the sample grows more ancient so intuitively makes sense with the increase in the jump rate that we observe under these conditions.</p>
<p>It also provided a useful intuition for how as the panel size grows, we are increasingly more able to capture more and more recent relatedness amongst tested haplotypes and the panel (e.g. you’re more and more likely to find a second cousin or closer relationship as the panel size increases). This is why phasing and imputation work really well in the current era of large sample sizes. In this derivation I was particularly interested in the point at which samples would be <strong>ancient enough</strong> that there is minimal difference in the coalescence time (suggesting a limit to imputation/phasing accuracy as well). Under a model of constant-size, we saw exactly this relationship, which is that fleetingly joyous intersection of intuition and validation that researchers are after!</p>
<!-- <figure class="figure">
	<img src="/images/blog_images/adna_ld/figureS17.png" width="500" height="400" alt="">
	<figcaption class="figcaption">Time to first coalescent event for an ancient sample with a lineage ancestral to the modern panel.</figcaption>
</figure><br>
 -->
<a class="reference internal image-reference" href="../../../_images/figureS17.png"><img alt="../../../_images/figureS17.png" class="align-center" src="../../../_images/figureS17.png" style="width: 500px; height: 400px;" />
</a>
<center>Time to first coalescent event for an ancient sample <br/> with a lineage ancestral to the modern panel.</center>
<p>If you’ve made it all the way down here and still find this work interesting, feel free to drop me a line via Twitter or email!</p>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          
          
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2025, Arjun Biddanda
            </div>
            Made with 
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            <div class="icons">
              <a class="muted-link " href="https://github.com/aabiddanda/" aria-label="GitHub">
            <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 16 16">
                <path fill-rule="evenodd" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"></path>
            </svg>
        </a>
              <a class="muted-link " href="https://bsky.app/profile/aabiddanda.bsky.social/" aria-label="Bluesky">
            <svg xmlns="http://www.w3.org/2000/svg" stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 576 512"><!--!Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2024 Fonticons, Inc.--><path d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3c3.4 .4 6.7 .9 10 1.3zM288 227.1C261.9 176.4 190.9 81.9 124.9 35.3C61.6-9.4 37.5-1.7 21.6 5.5C3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7c3.3-.5 6.6-.9 10-1.4c-3.3 .5-6.6 1-10 1.4C74.3 308.6-9.1 342.8 100.3 464.5C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4c102.4-103.4 28.1-156-65.8-169.9c-3.3-.4-6.7-.8-10-1.3c3.4 .4 6.7 .9 10 1.3c64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1z"/></svg>
            </svg>
        </a>
              <a class="muted-link " href="https://scholar.google.com/citations?user=BWVZXhgAAAAJ&hl=en" aria-label="Google Scholar">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--!Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2024 Fonticons, Inc.--><path d="M390.9 298.5c0 0 0 .1 .1 .1c9.2 19.4 14.4 41.1 14.4 64C405.3 445.1 338.5 512 256 512s-149.3-66.9-149.3-149.3c0-22.9 5.2-44.6 14.4-64h0c1.7-3.6 3.6-7.2 5.6-10.7c4.4-7.6 9.4-14.7 15-21.3c27.4-32.6 68.5-53.3 114.4-53.3c33.6 0 64.6 11.1 89.6 29.9c9.1 6.9 17.4 14.7 24.8 23.5c5.6 6.6 10.6 13.8 15 21.3c2 3.4 3.8 7 5.5 10.5zm26.4-18.8c-30.1-58.4-91-98.4-161.3-98.4s-131.2 40-161.3 98.4L0 202.7 256 0 512 202.7l-94.7 77.1z"/></svg>
        </a>
              
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Ancient DNA and LD</a><ul>
<li><a class="reference internal" href="#inspiration-and-starting-points">Inspiration and Starting Points</a></li>
<li><a class="reference internal" href="#the-impact-of-time-separation-on-genealogical-history-at-two-loci">The impact of time-separation on genealogical history at two loci</a></li>
<li><a class="reference internal" href="#copying-patterns-in-simulated-and-real-haplotype-data">Copying Patterns in Simulated and Real Haplotype Data</a></li>
<li><a class="reference internal" href="#outlook-future-work">Outlook &amp; Future Work</a></li>
<li><a class="reference internal" href="#personal-highlight">Personal Highlight</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="../../../_static/documentation_options.js?v=d4c685ee"></script>
    <script src="../../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/scripts/furo.js?v=5fa4622c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/tabs.js?v=3ee01567"></script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="https://www.googletagmanager.com/gtag/js?id=G-ZXEQFEKF68"></script>
    <script src="../../../_static/google_analytics_tracker.js?v=c15d0c3e"></script>
    </body>
</html>